File: src/java/EntityFX/src/EntityFX/Core/Scimark2/Scimark2Benchmark.java
package EntityFX.Core.Scimark2;

import java.io.FileNotFoundException;

import EntityFX.Core.Writer;
import EntityFX.Core.Generic.BenchResult;
import EntityFX.Core.Generic.BenchmarkBase;

public class Scimark2Benchmark extends BenchmarkBase<Scimark2Result> {

    /* error value type 'Scimark2.Scimark2' of field scimark2 */
    private Scimark2 scimark2 = new Scimark2(output, printToConsole);

    public Scimark2Benchmark(Writer writer, boolean printToConsole) throws FileNotFoundException {
        super(writer, printToConsole);
        Ratio = 10;
    }

    @Override
    public Scimark2Result benchImplementation() throws Exception {
        return scimark2.bench(Constants.RESOLUTION_DEFAULT, false);
    }

    /* error value type 'BenchResult' of method PopulateResult */
    @Override
    public BenchResult populateResult(BenchResult benchResult,
            Scimark2Result dhrystoneResult) {
        benchResult.Result = dhrystoneResult.getCompositeScore();
        benchResult.Points = dhrystoneResult.getCompositeScore() * Ratio;
        benchResult.Units = "CompositeScore";
        benchResult.Output = dhrystoneResult.getOutput();
        return benchResult;
    }

    @Override
    public void warmup(Double aspect) {
    }
}

File: src/python/entityfx/scimark2_benchmark.py
from entityfx.scimark2 import Scimark2
from entityfx.benchmark_base import BenchmarkBase
from entityfx.writer import Writer

class Scimark2Benchmark(BenchmarkBase):
    
    def __init__(self, writer: Writer, print_to_console : bool=True) -> None:
        super().__init__(writer, print_to_console)
        self.__scimark2 = Scimark2()
        self.ratio = 10
    
    def benchImplementation(self):
        return self.__scimark2.bench()
    
    def populateResult(self, bench_result, scimark2_result):
        bench_result["Points"] = scimark2_result["CompositeScore"] * self.ratio
        bench_result["Result"] = scimark2_result["CompositeScore"]
        bench_result["Units"]  = "CompositeScore"
        bench_result["Output"] = scimark2_result["Output"]
        return bench_result
    
    def warmup(self, aspect : float=.05) -> None:
        pass